flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -777,6 +781,16 @@ protected void convertSelectImpl(final Blackboard bb, SqlSelect select) {
 
         if (select.hasHints()) {
             final List<RelHint> hints = SqlUtil.getRelHint(hintStrategies, select.getHints());
+            if (!hints.isEmpty()) {
```

Comment:
Can `hints` be empty if `select.hasHints` says `true` ? If not, how about we move the config check before extracting the hints? It's unnecessary to go over the `RelNode` if we know we will throw an exception.

Reply:
Moved the config check before extracting the hints.